table/server/common/src/main/java/alluxio/table/common/udb/UdbMountSpec.java

```diff
@@ -0,0 +1,589 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.table.common.udb;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+/**
+ * Tables and partitions inclusion and exclusion specification.
+ */
+public final class UdbMountSpec {
+  /**
+   * Tables and partitions to bypass.
+   */
+  private final TablePartitionWrapper mBypassed;
+
+  /**
+   * Ignored tables.
+   * Tables are ignored as a whole, no partition configuration is needed.
+   */
+  private final SimpleWrapper mIgnored;
+
+  protected UdbMountSpec(TablePartitionWrapper bypassed,
+                         SimpleWrapper ignored) {
+    mBypassed = bypassed;
+    mIgnored = ignored;
+  }
+
+  /**
+   * Checks if a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be bypassed, false otherwise
+   * @see UdbMountSpec#hasFullyBypassedTable(String)
+   */
+  public boolean hasBypassedTable(String tableName) {
+    return !hasIgnoredTable(tableName) && mBypassed.hasTable(tableName);
```

Comment:
If the same table is configured to be ignored and bypassed at the same time, ignore should take precedence, I think.